### ISSUE TITLE
[$250] Make the avatar found in the shareable QR code the same that's found in the chat header

### ISSUE THREAD

**Issue by shawnborton** [Expensify/App#45876]
### Problem
When you go to view the Share Code of a chat report, ideally we should use the same exact avatar found in the chat header inside of the QR code itself. But right now we default to the Expensify iconmark logo in most cases. 

### Solution
Let's make it so that we always use the same avatar found within the chat header inside of the QR code. This also means making sure we are using the same shape too (circle vs rounded square). 

We already made some progress here as we now use the workspace's avatar inside of the workspace share code, but we need to extend this to all other chat reports (workspace rooms, IOUs, etc). 

cc @Kicu as you volunteered to take this one on. 
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~015e4ab1d74b325510</li>
        <li>Upwork Job ID: 1815295906307038940</li>
        <li>Last Price Increase: 2024-08-08</li>
<li>Automatic offers: </li>
<ul>
<li>rayane-djouah | Reviewer | 103212899</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @rayane-djouah</details>

**Comment by shawnborton** [comment]
I'm adding the External label here but note that we want to assign this job to @Kicu

**Comment by Kicu** [comment]
hey I will be working on this as I'm familiar with the code displaying QR codes :)

**Comment by Kicu** [comment]
yes, stuff happening in Search has took me more time than expected, will put more work here next week.

**Comment by Kicu** [comment]
yeah stil didn't start the work, but we're now close to finishing search advanced filters and in 1-2 days I should be able to do some work here.

**Comment by shawnborton** [comment]
The avatar should look exactly the same as it does in the chat header for all of those cases.

**Comment by shawnborton** [comment]
Sorry for the spam @dubielzyk-expensify 

@dylanexpensify can you make sure this one goes through our normal process here? cc @trjExpensify - not sure which project this should belong to.

**Comment by dylanexpensify** [comment]
Can do @shawnborton!

**Comment by Kicu** [comment]
I just wanted to add that I initially thought I will have time to take care of this one, however we are close to finishing Search v2.1 where I'm involved, so Im putting my work there for now.Perhaps if this task

**Comment by trjExpensify** [comment]
> cc @trjExpensify - not sure which project this should belong to.

Sheesh, me neither really. Happy to have it in #wave-collect general polish. 👍

**Comment by dylanexpensify** [comment]
Sounds good!

**Comment by shawnborton** [comment]
I don't think we'd be able to make a single SVG for each of those cases though, so I think we need to keep exploring solutions here.

**Comment by dylanexpensify** [comment]
bump @Kicu @rayane-djouah on an update

**Comment by Kicu** [comment]
Hey guys, sorry but the truth is for the past 2 weeks I have not done any work on this. 
I'm focusing on Search which is taking time and I haven't found any time for this task.

In general if this would be simple and doable within 1 day I would probably fix it. But this is a bigger task, because we never supported multiple svgs inside QR code in this app, and we can only support SVG at all because I added this support to 
`react-native-qrcode-svg` package. It will not support providing and styling multiple SVGs yet, so this issue will require multiple changes both to an external library and to our code - we can't simply take existing components and put them inside the generated QRCode.

I don't think I will find any time to work on this until we're done with Search.

**Comment by dylanexpensify** [comment]
Hi @Kicu thank you for the update! We appreciate the real take! Agree Search is higher priority here. @trjExpensify Do we want to look for another engineer to help with this (external), or are we good waiting until search is in a better spot to free up @Kicu's time?

**Comment by trjExpensify** [comment]
>and we can only support SVG at all because I added this support to react-native-qrcode-svg package. Right now this library will not support providing and styling multiple SVGs.

Is this an SWM library?

**Comment by Kicu** [comment]
@trjExpensify Since a few weeks it's actually yours 😅 the original owner passed the ownership to Expensify org: https://github.com/Expensify/react-native-qrcode-svg/ CC @grgia @luacmartins 

However this lib is used by a lot of people, so I think changes to its public api should be thought through.
A while back I added the ability to display SVGs inside the QRcode, because previously it only accepted raster images, via URL.
This added an extra two props to it, and now it is able to display an svg logo either by url or by passing the SVG code directly.

**tl;dr**
It is easy and doable right now is to display an image inside QRcode - either jpg/png or SVG.
What requires modifications both in App and in `react-native-qrcode-svg` would be displaying any combinations looking like this:
<p>
<img width="122" alt="Screenshot 2024-09-20 at 11 24 08" src="https://github.com/user-attachments/assets/74882256-e4ce-446b-8891-b441ab69d9ca">
<img width="88" alt="Screenshot 2024-09-20 at 11 28 49" src="https://github.com/user-attachments/assets/211546a1-c893-4077-aeef-e1bbe8f98b91">
</p>

**Comment by Kicu** [comment]
PR that added SVG implementation: https://github.com/Expensify/react-native-qrcode-svg/pull/207

**Comment by trjExpensify** [comment]
>@trjExpensify Since a few weeks it's actually yours 😅 the original owner passed the ownership to Expensify org: https://github.com/Expensify/react-native-qrcode-svg/

Ah, nice. I didn't know that!  

>What is harder and requires modifications both in App and in react-native-qrcode-svg would be displaying any combination looking like this:

Maybe a noob question, but why can't we combine the dual avatar into one SVG for the QR code? I'm not sure I followed that part.

**Comment by Kicu** [comment]
> Maybe a noob question, but why can't we combine the dual avatar into one SVG for the QR code?
This is a great question with a tricky answer 😅

It depends on what is your definition of `combine`.

Let's assume we have 2 SVGs: `Avatar1.svg` and `Avatar2.svg`.
How we use them in the project is we simply import them, then during import some webpack (or babel...) plugin runs over the files and gives us the same svg as a react-component - which is easier to use.
```ts
import Avatar1 from ".../assets/.../Avatar1.svg`
[..]

return <View>
   <Avatar1 {...} />...
```
(Here is an example of our default avatar from App: https://github.com/Expensify/App/blob/main/assets/images/avatars/user/default-avatar_1.svg?short_path=971389c)

The `react-native-qrcode-svg` package (at this point) only allows me to pass it 1 SVG asset, either as: an URL, a svg string or svg wrapped in react component. 
So to combine my `Avatar1.svg` and `Avatar2.svg` into one, I would have to unpack (parse) their contents and somehow glue them together into 1 svg object. This seems complex and inconvenient. (don't forget we would have to make one of them smaller than the other).
Alternatively perhaps we could wrap the two together into another react component, that simply returns both. But still we need to apply extra styling on them and position them relative to each other.

In addition there are basically limitless combinations of these cases:
 - user SVG default avatar
 - workspace SVG default avatar
 - user/workspace uploaded avatar (which could be `.png` or other raster)
 
That is why I'm calling this task a complex one.
But again this is only basic answer, since I didn't do a deeper dive. I'm sure this is doable, just might be a bit time consuming.

**Comment by trjExpensify** [comment]
>But again this is only basic answer, since I didn't do a deeper dive. I'm sure this is doable, just might be a bit time consuming.

Okay cool, thanks for the explanation. Shawn is out for a couple of weeks, but @dannymcclain @dubielzyk-expensify how motivated are we for this change soon? @Kicu is tied up with Search stuff, so if we think it's important, we'll need to see if we can loop in someone else to progress it probably.

**Comment by dannymcclain** [comment]
>how motivated are we for this change soon?

Curious how Jon feels—I think this is something we need to do but it doesn't feel SUPER DUPER important to me.

**Comment by dubielzyk-expensify** [comment]
Agree. Let's do this after Search cause that's more important to me. But I'd love if we could not lose track of it and quickly follow with it.

**Comment by dubielzyk-expensify** [comment]
Yep

**Comment by shawnborton** [comment]
Hey, I'm back! What's needed from me here?

**Comment by dylanexpensify** [comment]
Reassigning as I head to parental leave next week! Thanks Rach!

**Comment by RachCHopkins** [comment]
I'm well out of the loop here @shawnborton @dubielzyk-expensify @dannymcclain  - Jon mentioned it being a priority once Search is up and running, are we at that stage yet do you think?

**Comment by dubielzyk-expensify** [comment]
My gut says we still shouldn't tackle this at this moment, but happy to hear what S and D has to say

**Comment by shawnborton** [comment]
Definitely low priority compared to all of the other things that are happening, but is it something that can be tackled 100% by our contributor community?

**Comment by dannymcclain** [comment]
>My gut says we still shouldn't tackle this at this moment
>Definitely low priority compared to all of the other things that are happening

Agree with Jon and Shawn. Definitely not super high priority right now.

>is it something that can be tackled 100% by our contributor community?

Also curious about this.

**Comment by Kicu** [comment]
> is it something that can be tackled 100% by our contributor community?

The only possible problem I see is that this most likely will require changes in another open source library we are using: https://github.com/Expensify/react-native-qrcode-svg (which Expensify is maintaining btw ;))

I think at this moment this library will not support passing multiple SVG images to it, so this issue would require work in both `Expensify/App` and `Expensify/react-native-qrcode-svg`

I'm still busy doing Search, handling all the edge cases in autocomplete so I still don't wanna tackle this yet.

**Comment by RachCHopkins** [comment]
Ok, let's close this for now and it can be opened again when ready.
